draft-ietf-ppm-dap.md

```diff
@@ -1861,9 +1894,9 @@ scenario where the Helper successfully advances from step `n` to `n+1`, but its
 `AggregationJobResp` response to the Leader gets dropped due to something like a
 transient network failure. The Leader could then resend the request to have the
 Helper advance to step `n+1` and the Helper should be able to retransmit the
-`AggregationJobContinueReq` that was previously dropped. To make that kind of
-recovery possible, Aggregator implementations SHOULD checkpoint the most recent
-step's prep state and messages to durable storage such that the Leader can
+`AggregationJobResp` that was previously dropped. To make that kind of recovery
+possible, Aggregator implementations SHOULD checkpoint the most recent step's
```

Comment:
Can we make this plural to indicate that Helpers can store anywhere from the most recent step to the all the steps in a job's history?

Reply:
Clarify? The Helper can store whatever it wants, but the minimum recommendation is to store whatever it needs to satisfy skew recovery, as far as the protocol is concerned.

Reply:
The language implies that only the most recent step's state SHOULD be stored. I think we can generalize this language to imply that _all_ steps' state SHOULD/MAY be stored so promote aggregation step skew recovery.

Reply:
Or in other words, the Helper could retain the entire step history and satisfy the SHOULD, but I'm not sure that possibility needs to be explicitly called out in protocol text.

Reply:
In general, the aggregation step skew recovery section seems to focus on just recovering when there's a difference of 1 between the Helper and Leader. I think it would be beneficial to change the wording to imply that aggregation step skew recovery can allow the Leader to re-send any number of previously computed steps. Hence, specifying the Helper can store all previous steps' state follows naturally from this.

Reply:
> The language implies that only the most recent step's state SHOULD be stored.

I don't think I agree with that reading, but we could say "Aggregator implementations SHOULD, at minimum, checkpoint...".

(n.b. this diff got roped into this PR because I noticed a typo in `AggregationJobResp`. We may consider taking this editorial into another PR).

Reply:
> I think it would be beneficial to change the wording to imply that aggregation step skew recovery can allow the Leader to re-send any number of previously computed steps. Hence, specifying the Helper can store all previous steps' state follows naturally from this.

Ah, this is the part I was missing, thanks for clarifying. I do think we should rope this into another PR though, since I think there could be more wording changes necessary to make this clear. For one we only call out recovering from `n` to `n+1`.

Reply:
Opened a new PR on this edit: https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/pull/569
We can resolve this thread here 😁